program-runtime/src/loaded_programs.rs

```diff
@@ -428,6 +435,19 @@ impl LoadedProgram {
             && slot >= self.deployment_slot
             && slot < self.effective_slot
     }
+
+    pub fn update_access_slot(&self, slot: Slot) {
+        let mut current_slot = self.latest_access_slot.write().unwrap();
```

Comment:
could drop the lock in favor of another `AtomicU64`. this method is then a loop on `compare_exchange` while `new < current`

Reply:
I had first done it with `AtomicU64`, and then decided to use the absolute type for `Slot`. I don't think we ever will change `Slot` to not being a `u64`. So maybe it's ok to use `AtomicU64`.